src/coreclr/pgosupport.cmake

```
@@ -6,7 +6,7 @@ if(NOT WIN32)
   # Function required to give CMAKE_REQUIRED_* local scope
   function(check_have_lto)
     set(CMAKE_REQUIRED_FLAGS -flto)
-    set(CMAKE_REQUIRED_LIBRARIES -flto -fuse-ld=gold)
+    set(CMAKE_REQUIRED_LIBRARIES -flto -fuse-ld=lld)
```

We don't need to re-specify linker flavor here, it is inserted globally:
```suggestion
    set(CMAKE_REQUIRED_LIBRARIES -flto)
```
(unless there was a reason to keep `gold` linker here, in which case, undo)